src/frequenz/sdk/microgrid/_power_distributing/_component_managers/_pv_inverter_manager/_pv_inverter_manager.py

```diff
@@ -263,3 +284,69 @@ def _get_pv_inverter_ids(self) -> collections.abc.Set[int]:
             )
             if inv.type == InverterType.SOLAR
         }
+
+    def _get_connected_meters(
+        self, pv_inverter_ids: collections.abc.Set[int]
+    ) -> collections.abc.Mapping[int, int]:
+        """Return the connected meters for the given PV inverters.
+
+        Args:
+            pv_inverter_ids: The PV inverter ids to get the connected meters for.
+
+        Returns:
+            A dictionary mapping the PV inverter ids to the connected meter ids.
+        """
+        component_graph = connection_manager.get().component_graph
+        ret = {}
+        for inv_id in pv_inverter_ids:
+            predecessors = component_graph.predecessors(inv_id)
+            if len(predecessors) != 1:
+                continue
+
+            predecessor = predecessors.pop()
+            if predecessor.category != ComponentCategory.METER:
+                continue
+
+            if len(component_graph.successors(predecessor.component_id)) != 1:
+                continue
+
+            ret[inv_id] = predecessor.component_id
+        return ret
+
+    def _get_unreachable_inv_power(
+        self, request: Request, working_components: list[int]
+    ) -> Power:
+        """Return the power of the unreachable inverters in the request.
+
+        Args:
+            request: The request to get the unreachable inverters power for.
+            working_components: The working components in the request.
+
+        Returns:
+            The power of the unreachable inverters in the request.
+        """
+        unreachable_inv_ids = request.component_ids - set(working_components)
+        if not unreachable_inv_ids:
+            return Power.zero()
+        unreachable_power = Power.zero()
+        now = datetime.now(tz=timezone.utc)
+        for inv_id in unreachable_inv_ids:
+            if inv_id not in self._connected_meters:
+                unreachable_power += self._fallback_power
+                continue
+
+            meter_id = self._connected_meters[inv_id]
+            meter_cache = self._meter_data_caches[meter_id]
+            if not meter_cache.has_value():
+                unreachable_power += self._fallback_power
+                continue
+
+            meter_data = meter_cache.get()
+            if now - meter_data.timestamp > MAX_DATA_AGE:
+                unreachable_power += self._fallback_power
+                continue
+
+            unreachable_power = Power.from_watts(
+                unreachable_power.as_watts() + meter_data.active_power
+            )
+        return unreachable_power
```

Comment:
Not sure but from logs I saw components are changing state from not-working to working after  ~10sec . 
That would mean that every 10 sec we would try to distribute whole requested pv power - and (probably) feed -in.

Maybe we should check last message timestamp for all meters connected to the requested inverters - not only the unreachable one ?

Reply:
If components are not sending data, their state doesn't change to `WORKING` until there's data again.  Automatic change only happens from `UNCERTAIN` to `WORKING`, never from `NOT_WORKING` to `WORKING`.  (And `UNCERTAIN` happens when there's data and no error state in the data, but a request to the API has failed, and `NOT_WORKING` happens 10 seconds after there's no data or immediately after data shows component is in error state.)

I've also tested this code and that's not an issue.